rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -51,6 +51,12 @@ paths:
                   format: binary
                   minLength: 1
                   maxLength: 14316557655
+                correlationId:
+                  description: >
+                    A unique ID that is passed by an external client that can be used to correlate a product
+                    stored in ION with the same product stored in the external client's system.
+                  type: string
+                  minLength: 1
```

Comment:
Should this have a max length?

Reply:
I was about to leave a comment asking for suggestions since this could possibly be a non-standard form of ID

Reply:
Here's a suggestion:
```yaml
schema:
  type: string
  minLength: 32
  maxLength: 32
  pattern: '[^A-Za-z0-9]+'
```

Reply:
I don't think we should enforce the pattern here. We do not now what is the format of the systems replication might be getting this id form. As for maximum length, we should maybe bump this up as again, we do not know the format of the system we might be harvesting from (although we do know for DIB right now). In any cases, most databases today don't reserve the whole length of a string based on its max and therefore I would suggest putting the limit higher.

Reply:
✏️ I would remove the word `external` in the above description. From CDR's perspective, the ID is provided to them by the client of this API. I would also not talk about how replication is planning on using it but more on what is guaranteed by CDR with this id

❗️ I think the API also need to document somehow as part of its contract that the implementor of this API will guarantee that only one entry in the database will ever exist for a given correlation id.

Reply:
@paouelle, to speed up the review process, can you provide the code for the schema and description that you're expecting, please?

Reply:
I would remove the pattern altogether and maybe fix the maxLength to 255 for now.
As for the description, how about something like:
```
A unique client provided identifier for the product to be ingested which will allow (in the future) the client to retrieve the product and/or any other information associated with it using this correlation identifier. Only one product can exist with a given correlation identifier which means that a request to ingest a product with an existing correlation id will result in XXXX.
```

`XXXX` being the behavior that is currently expected. In my opinion, since we do not support update at the moment, this behavior should be `a failure with no changes to the existing product or associated information`.